src/language-js/needs-parens.js

```diff
@@ -488,35 +495,32 @@ function needsParens(path, options) {
       }
 
       return (
+        name === "object" &&
         parent.type === "MemberExpression" &&
-        typeof node.value === "number" &&
-        name === "object"
+        typeof node.value === "number"
       );
 
     case "AssignmentExpression": {
       const grandParent = path.getParentNode(1);
 
-      if (parent.type === "ArrowFunctionExpression" && parent.body === node) {
+      if (name === "body" && parent.type === "ArrowFunctionExpression") {
         return true;
       } else if (
+        name === "key" &&
         parent.type === "ClassProperty" &&
-        parent.key === node &&
         parent.computed
       ) {
         return false;
-      } else if (
-        parent.type === "TSPropertySignature" &&
-        parent.name === node
-      ) {
```

Comment:
This seems to be an artifact left from older versions of `typescript-eslint`. `TSPropertySignature` doesn't have the `name` property anymore. See https://github.com/eslint/typescript-eslint-parser/commit/5421ceae59a10212562eedbafbd4bea116ca2b6e

Reply:
Nice catch, I thought tests not covered.